docs/data/data-grid/server-side-data/ServerSideRowGroupingDataGrid.js

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  useGridApiRef,
+  useKeepGroupedColumnsHidden,
+} from '@mui/x-data-grid-premium';
+import { useMockServer } from '@mui/x-data-grid-generator';
+import Button from '@mui/material/Button';
+
+export default function ServerSideRowGroupingDataGrid() {
+  const apiRef = useGridApiRef();
+
+  const { fetchRows, columns } = useMockServer({
+    rowGrouping: true,
+  });
+
+  const dataSource = React.useMemo(() => {
+    return {
+      getRows: async (params) => {
+        const urlParams = new URLSearchParams({
+          paginationModel: JSON.stringify(params.paginationModel),
+          filterModel: JSON.stringify(params.filterModel),
+          sortModel: JSON.stringify(params.sortModel),
+          groupKeys: JSON.stringify(params.groupKeys),
+          groupFields: JSON.stringify(params.groupFields),
+        });
+        const getRowsResponse = await fetchRows(
+          `https://mui.com/x/api/data-grid?${urlParams.toString()}`,
+        );
+        return {
+          rows: getRowsResponse.rows,
+          rowCount: getRowsResponse.rowCount,
+        };
+      },
+      getGroupKey: (row) => row.group,
+      getChildrenCount: (row) => row.descendantCount,
+    };
+  }, [fetchRows]);
+
+  const initialState = useKeepGroupedColumnsHidden({
+    apiRef,
+    initialState: {
+      rowGrouping: {
+        model: ['company', 'director'],
```

Comment:
Grouping by `year` doesn't seem to work in this demo.
Is this also related to the mock server?

Reply:
Yes, [fixed](https://github.com/mui/mui-x/pull/13826/commits/1b5bd369df398b418867f0a4bf47e73c5e402bcf).